enumerable.rb

```diff
@@ -0,0 +1,200 @@
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    i = 0
+    select_array = []
+    while i < size
+      select_array << yield(self[i])
+      i += 1
+    end
+    select_array
+  end
+
+  def my_all?(pattern = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    result = true
+    current = self
+    if !block_given? && pattern.nil?
+      current.my_each do |item|
+        result = false unless item
+      end
+    elsif pattern.is_a? Regexp
+      current.my_each do |item|
+        result = item =~ pattern
+      end
+    elsif pattern.is_a? Class
+      current.my_each do |item|
+        result = false unless item.is_a? pattern
+      end
+    elsif pattern
+      current.my_each do |item|
+        result = false unless item == pattern
+      end
+    elsif is_a? Array
+      current.my_each do |item|
+        result = false unless yield(item)
+      end
+    elsif is_a? Hash
+      current.my_each do |k, v|
+        result = false unless yield(k, v)
+      end
+    end
+    result
+  end
+
+  def my_any?(pattern = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    current = self
+    result = false
+    if !block_given? && pattern.nil?
+      current.my_each do |item|
+        result = true if item
+      end
+    elsif pattern.is_a? Regexp
+      current.my_each do |item|
+        result = true if item =~ pattern
+      end
+    elsif pattern.is_a? Class
+      current.my_each do |item|
+        result = true if item.is_a? pattern
+      end
+    elsif pattern
+      current.my_each do |item|
+        result = true if item == pattern
+      end
+    elsif is_a? Array
+      current.my_each do |item|
+        result = true if yield(item)
+      end
+    elsif is_a? Hash
+      current.my_each do |k, v|
+        result = true if yield(k, v)
+      end
+    end
+    result
+  end
+
+  def my_none?(pattern = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    result = true
+    current = self
+    if !block_given? && pattern.nil?
+      current.my_each do |item|
+        result = false if item
+      end
+    elsif pattern.is_a? Class
+      current.my_each do |item|
+        result = false if item.is_a? pattern
+      end
+    elsif pattern.is_a? Regexp
+      current.my_each do |item|
+        result = false if item =~ pattern
+      end
+    elsif pattern.is_a? Array
+      current.my_each do |item|
+        result = false if yield(item)
+      end
+    elsif pattern
+      current.my_each do |item|
+        result = false if item == pattern
+      end
+    end
+    result
+  end
+
+  def my_count(parameter = nil)
+    count = 0
+    if block_given?
+      my_each { |i| count += 1 if yield(self[i]) }
+    elsif !parameter.nil?
+      my_each { |i| count += 1 if self[i] == parameter }
+    else
+      count = length
+    end
+    count
+  end
```

Comment:
- [ ] Enumerables `my_count` counts the number of items in enum that are equal to the item if an argument is given
`[1,2,3,4,5].my_count(3)` should return `1`

- [ ] Enumerables `my_count` counts the number of elements yielding a true value if a block is given
`[1,2,3,4,5].my_count{ |num| num == 4 }` should return `1`

Reply:
these tests are also passing